File: TociHaia/Toci.Haia.Studio.Web/src/features/meme-intakes/contracts/memeIntakeContracts.ts
export type CreateMemeIntakeRequest = {
	fileName: string
	contentType: string
	sizeBytes: number
	workingTitle?: string
}

export type CreateMemeIntakeResponse = {
	intakeId: string
	candidateId: string
	candidateVersionId: string
	assetId: string
	objectKey: string
	uploadUrl: string
	uploadHeaders: Record<string, string>
	uploadUrlExpiresAtUtc: string
	status: string
}

export type FinalizeMemeIntakeResponse = {
	intakeId: string
	status: string
	contentType: string
	sizeBytes: number
	sha256Hash: string
}

export type MemeClassification = {
	axisKey: string
	axisDisplayName: string
	valueKey: string
	valueDisplayName: string
	relevanceScore: number
	confidence: number
	isPrimary: boolean
	rankNo?: number | null
}

export type MemeMeasure = {
	axisKey: string
	axisDisplayName: string
	normalizedValue: number
	confidence: number
}

export type MemeSafety = {
	categoryKey: string
	categoryDisplayName: string
	severityLevel: number
	confidence: number
	moderationRelevance: boolean
}

export type MemeReaction = {
	text: string
	displayOrder: number
	initiallyVisible: boolean
	relevanceScore: number
	confidence: number
	secondPunchline: string
	mechanismKeys: string[]
}

export type MemeEvaluationResult = {
	suggestedTitle: string
	visualDescription: string
	detectedText?: string | null
	languageCode: string
	contentFormat: string
	suggestedRoleInFlow: string
	suggestedRoleConfidence: number
	editorialSummary?: string | null
	overallConfidence: number
	classifications: MemeClassification[]
	measures: MemeMeasure[]
	safety: MemeSafety[]
	moderationRecommendation: string
	moderationSummary?: string | null
	predictedDrynessLevel: number
	predictedDrynessLabel: string
	predictedDrynessConfidence: number
	reactions: MemeReaction[]
}

export type EvaluateMemeIntakeResponse = {
	intakeId: string
	status: string
	evaluation: MemeEvaluationResult
}

export type StartEditorialReviewResponse = {
	intakeId: string
	status: string
	editorialStatus: string
	classificationStatus: string
	candidateClassificationId?: string | null
}

export type MemeEditorialDecisionRequest = {
	summary?: string | null
	editorialNote?: string | null
}

export type MemeEditorialDecisionResponse = {
	intakeId: string
	status: string
	editorialStatus: string
	classificationStatus: string
	reviewStatus: string
	candidateClassificationId: string
	reviewedAtUtc: string
	reviewedByAccountId: string
}

export type GetMemeIntakeResponse = {
	intakeId: string
	status: string
	workingTitle?: string | null
	asset: {
		fileName: string
		contentType: string
		sizeBytes: number
		objectKey: string
	}
	evaluation?: MemeEvaluationResult | null
}

File: TociHaia/Toci.Haia.AiIntegration.Core/Toci.Haia.Studio.Api/Features/MemeIntakes/MemeIntakeContracts.cs
namespace Toci.Haia.Studio.Api.Features.MemeIntakes;

public sealed record CreateMemeIntakeRequest(
    string FileName,
    string ContentType,
    long SizeBytes,
    string? WorkingTitle);

public sealed record CreateMemeIntakeResponse(
    Guid IntakeId,
    Guid CandidateId,
    Guid CandidateVersionId,
    Guid AssetId,
    string ObjectKey,
    string UploadUrl,
    IReadOnlyDictionary<string, string> UploadHeaders,
    DateTimeOffset UploadUrlExpiresAtUtc,
    string Status);

public sealed record FinalizeMemeIntakeResponse(
    Guid IntakeId,
    string Status,
    string ContentType,
    long SizeBytes,
    string Sha256Hash);

public sealed record EvaluateMemeIntakeResponse(
    Guid IntakeId,
    string Status,
    MemeEvaluationResultDto Evaluation);

public sealed record StartEditorialReviewResponse(
    Guid IntakeId,
    string Status,
    string EditorialStatus,
    string ClassificationStatus,
    Guid? CandidateClassificationId);

public sealed record MemeEditorialDecisionRequest(
    string? Summary,
    string? EditorialNote);

public sealed record MemeEditorialDecisionResponse(
    Guid IntakeId,
    string Status,
    string EditorialStatus,
    string ClassificationStatus,
    string ReviewStatus,
    Guid CandidateClassificationId,
    DateTimeOffset ReviewedAtUtc,
    Guid ReviewedByAccountId);

public sealed record GetMemeIntakeResponse(
    Guid IntakeId,
    string Status,
    MemeAssetDto Asset,
    MemeEvaluationResultDto? Evaluation,
    string? WorkingTitle);

public sealed record MemeAssetDto(
    string FileName,
    string ContentType,
    long SizeBytes,
    string ObjectKey);

public sealed record MemeEvaluationResultDto
{
    public required string SuggestedTitle { get; init; }

    public required string VisualDescription { get; init; }

    public string? DetectedText { get; init; }

    public required string LanguageCode { get; init; }

    public required string ContentFormat { get; init; }

    public required string SuggestedRoleInFlow { get; init; }

    public decimal SuggestedRoleConfidence { get; init; }

    public string? EditorialSummary { get; init; }

    public decimal OverallConfidence { get; init; }

    public required IReadOnlyList<MemeClassificationDto> Classifications { get; init; }

    public required IReadOnlyList<MemeMeasureDto> Measures { get; init; }

    public required IReadOnlyList<MemeSafetyDto> Safety { get; init; }

    public required string ModerationRecommendation { get; init; }

    public string? ModerationSummary { get; init; }

    public int PredictedDrynessLevel { get; init; }

    public required string PredictedDrynessLabel { get; init; }

    public decimal PredictedDrynessConfidence { get; init; }

    public required IReadOnlyList<MemeReactionDto> Reactions { get; init; }
}

public sealed record MemeClassificationDto(
    string AxisKey,
    string AxisDisplayName,
    string ValueKey,
    string ValueDisplayName,
    decimal RelevanceScore,
    decimal Confidence,
    bool IsPrimary,
    int? RankNo);

public sealed record MemeMeasureDto(
    string AxisKey,
    string AxisDisplayName,
    decimal NormalizedValue,
    decimal Confidence);

public sealed record MemeSafetyDto(
    string CategoryKey,
    string CategoryDisplayName,
    short SeverityLevel,
    decimal Confidence,
    bool ModerationRelevance);

public sealed record MemeReactionDto(
    string Text,
    int DisplayOrder,
    bool InitiallyVisible,
    decimal RelevanceScore,
    decimal Confidence,
    string SecondPunchline,
    IReadOnlyList<string> MechanismKeys);
